### DISCUSSION TITLE
Does Z-Wave JS and any devices feature power grid frequency Hz sensor as part of mains-power energy meters?

### DISCUSSION THREAD

**Discussion by Hedda** [zwave-js/zwave-js#3695] [Q&A]
Electricity "frequency" in Hertz as in frequency sensor mains-power grid is suggested as a new device class here in Home Assistant:

https://github.com/home-assistant/architecture/discussions/600

https://github.com/home-assistant/core/pull/59417

For example, the grid frequency aims to be around 60Hz ±1% in North America, 50Hz ±1% in the European Union, so it can fluctuate.

The question is whether any existing Z-Wave devices offer a power grid "frequency" sensor and if so does Z-Wave JS support it?

Maybe some clamp power meter or other energy meters?

Aeotec Home Energy Meter and Qubino 3-Phase Smart Meter does for example not specifically list a frequency sensor, ... but maybe?

https://aeotec.com/z-wave-home-energy-measure/index.html

https://qubino.com/products/3-phase-smart-meter/

**Comment by AlCalzone** [comment]
All I can say is that there exists a scale in in the specs for Frequency in Hz or KHz.

**Comment by Hedda** [comment]
FYI @epenet has now submitted pull requests for adding "frequency" (Hz, MHz, or GHz) as a new device class in Home Assistant core and UI:

https://github.com/home-assistant/core/pull/59700

https://github.com/home-assistant/frontend/pull/10621

https://github.com/home-assistant/developers.home-assistant/pull/1137

https://github.com/home-assistant/home-assistant.io/pull/20319

More discussion here:

https://github.com/home-assistant/architecture/discussions/600

epenet pointed however pointed out that `kHz` is not (yet?) part of the standard units in const.py of Home Assistant core:

https://github.com/home-assistant/core/blob/5370dd81226853c7c8e6b2c5b7310c38a3e0948a/homeassistant/const.py#L472-L475

```
 # Frequency units 
 FREQUENCY_HERTZ: Final = "Hz" 
 FREQUENCY_MEGAHERTZ: Final = "MHz" 
 FREQUENCY_GIGAHERTZ: Final = "GHz"
```

Not sure if "kHz" can just added as an additional standard unit in const.py for PR https://github.com/home-assistant/core/pull/59700 or if a new HA architecture discussion and decision needed specifically for that(?) https://github.com/home-assistant/architecture/discussions/ as per https://github.com/home-assistant/architecture/discussions/494


```
 FREQUENCY_KILOHERTZ: Final = "kHz"
```

Unit for `kHz` is needed to stay compliant with Z-Wave specification without having to do the conversion in the Z-Wave JS integration, or?

**Reply by raman325** [reply]
Guess I have good timing: https://github.com/home-assistant/core/pull/59659

As part of that I added `FREQUENCY_KILOHERTZ`

**Reply by Hedda** [reply]
> Guess I have good timing: [home-assistant/core#59659](https://github.com/home-assistant/core/pull/59659)
> 
> As part of that I added `FREQUENCY_KILOHERTZ`

Nice! It was originally added now as a dependency Frequency and Power Factor scales for ZHA -> https://github.com/home-assistant/core/pull/59417

Might want to give a short thanks to epenet for quickly adding freqency here -> https://github.com/home-assistant/architecture/discussions/600
